src/platform-aws.c

```diff
@@ -158,38 +126,58 @@ static struct ec2_platform_data platform_data_map[] = {
  * @return	NULL, if no topology found
  * 		platform data, if match found
  */
-static struct ec2_platform_data *get_platform_data()
+static struct ec2_platform_data *get_platform_data(void)
 {
 	static bool init = false;
 	static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
 	static struct ec2_platform_data *platform_data = NULL;
 	const size_t platform_n = sizeof(platform_data_map)/sizeof(platform_data_map[0]);
 	const char* platform_type = NULL;
+	regex_t regex;
+	int ret;
 
 	nccl_net_ofi_mutex_lock(&mutex);
 
 	if (init) {
-		nccl_net_ofi_mutex_unlock(&mutex);
-		return platform_data;
+		goto done;
 	}
 	init = true;
 
 	platform_type = nccl_net_ofi_get_product_name();
 	if (platform_type == NULL) {
-		nccl_net_ofi_mutex_unlock(&mutex);
-		return NULL;
+		goto done;
 	}
 
 	for (size_t idx = 0; idx < platform_n; idx++) {
-		if (strcmp(platform_type, platform_data_map[idx].name) == 0)
+		ret = regcomp(&regex, platform_data_map[idx].name, 0);
+		if (ret != 0) {
+			NCCL_OFI_WARN("Could not compile platform_type regex for %s",
+				      platform_data_map[idx].name);
+			goto done;
+		}
+
+		ret = regexec(&regex, platform_type, 0, NULL, 0);
+
+		regfree(&regex);
+
+		if (ret == 0) {
 			platform_data = &platform_data_map[idx];
+			break;
+		} else if (ret != REG_NOMATCH) {
+			NCCL_OFI_WARN("Regex match failed");
```

Comment:
Really an `unexpected error` rather than match failure? The man page suggests `regexec` won't return anything but `0` or `REG_NOMATCH`.